api/pkg/apps/attachments/interface.go

```diff
@@ -0,0 +1,22 @@
+package attachments
+
+import "net/url"
+
+/*type Interface interface {
+	Attachments() AttachmentClient
+}*/
+
+type AttachmentListOptions struct {
+	AttachedToID string
+}
+
+func (a *AttachmentListOptions) MarshalQueryParameters() (url.Values, error) {
+	ret := url.Values{}
+	ret.Set("attachedToId", a.AttachedToID)
+	return ret, nil
+}
+
+func (a *AttachmentListOptions) UnmarshalQueryParameters(values url.Values) error {
+	a.AttachedToID = values.Get("attachedToId")
+	return nil
+}
```

Comment:
What's the reason for placing these in this file? I don't think they are part of the package's interface. If anything they should be private methods (lowercase first letter)

Reply:
I was inspired by this file: https://github.com/nrc-no/core/blob/74d0e2f6c2080d5fdfa656bf5e6702992a9135b9/api/pkg/apps/iam/client_interface.go

Should I change them to being private?

Reply:
I could also move the Marshal/Unmarshal to a separate file, what do you suggest?

Reply:
Yeah, there are lots of methods/functions all over the place which I think should be private. If they are private they could stay I think. Otherwise maybe they should go somewhere else. Other option: rename the file :P ..after all the package's interface goes beyond what's in this file